vermouth/pdb/cif.py

```diff
@@ -0,0 +1,133 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Provides functionality to read CIF files.
+"""
+
+try:
+    from CifFile import ReadCif
+except ImportError:
+    HAVE_READCIF = False
+else:
+    HAVE_READCIF= True
+
+import numpy as np
+
+from ..molecule import Molecule
+
+def _cell(cf, modelname):
+    """
+    get the cell dimensions from a cif file as a list
+    """
+    a = cf[modelname]['_cell.length_a']
+    b = cf[modelname]['_cell.length_b']
+    c = cf[modelname]['_cell.length_c']
+    alpha = cf[modelname]['_cell.angle_alpha']
+    beta = cf[modelname]['_cell.angle_beta']
+    gamma = cf[modelname]['_cell.angle_gamma']
+
+    return np.array([a, b, c, alpha, beta, gamma], dtype=np.float32)
+
+def read_cif_file(file_name, exclude=('SOL', 'HOH'), ignh=False):
+    """
+    Parse a CIF file to create a molecule using the PyCIFRW library
+
+    Parameters
+    ----------
+    filename: str
+        The file to read.
+    exclude: collections.abc.Container[str]
+        Atoms that have one of these residue names will not be included.
+    ignh: bool
+        Whether hydrogen atoms should be ignored.
+    model: int
+        If the PDB file contains multiple models, which one to select.
+
+    Returns
+    -------
+    list[vermouth.molecule.Molecule]
+        The parsed molecules. Will only contain edges if the PDB file has
+        CONECT records. Either way, the molecules might be disconnected. Entries
+        separated by TER, ENDMDL, and END records will result in separate
+        molecules.
+    """
+
+    # list of categories from the read cif file to read into a molecule
+    cif_categories = [
+    '_atom_site.id', # atomid
+    '_atom_site.auth_atom_id', # atomname
+    '_atom_site.auth_comp_id', # resname
+    '_atom_site.auth_asym_id', # chain
+    '_atom_site.auth_seq_id', # resid
+    #'_atom_site.pdbx_pdb_ins_code', # insertion_code
+    '_atom_site.cartn_x', # x
+    '_atom_site.cartn_y', # y
+    '_atom_site.cartn_z', # z
+    #'_atom_site.occupancy', # occupancy
+    #'_atom_site.b_iso_or_equiv', # temp_factor
+    '_atom_site.type_symbol', # element
+    '_atom_site.pdbx_formal_charge', # charge
+    ]
+    cif_category_names = ['atomid', 'atomname', 'resname', 'chain', 'resid',
+                          'x', 'y', 'z',
+                          'element', 'charge']
+    cif_category_types = [int, str, str, str, int,
+                          float, float, float,
+                          str, str]
+
+    cf = ReadCif(str(file_name))
+
+    molecules = []
+
+    for model in cf.keys():
+
+        molecule = Molecule()
+        # annoyingly _refine_hist.number_atoms_total is not a mandatory entry
+        number_atoms = len(cf[model]['_atom_site.cartn_x'])
```

Comment:
```suggestion
        number_atoms = len(cf[model]['_atom_site.id'])
```
Is it worth checking whether _refine_hist.number_atoms_total is the same as number_atoms? (if given)